src/CoreWCF.NetTcp/tests/TransportWithMessageCredentialNetTCPTest.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.ConstrainedExecution;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.ServiceModel.Channels;
+using CoreWCF.Configuration;
+using CoreWCF.IdentityModel.Selectors;
+using CoreWCF.Security;
+using Helpers;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace CoreWCF.NetTcp.Tests
+{
+    public class TransportWithMessageCredentialNetTCPTest
+    {
+        private readonly ITestOutputHelper _output;
+
+        public TransportWithMessageCredentialNetTCPTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact ]
+        public void BasicUserNameAuth()
+        {
+            runTest(false, "testuser@corewcf", _output);
+        }
+
+        [Fact]
+        public void BasicUserNameAuthError()
+        {
+            runTest(true, "randomuser@corewcf", _output);
+        }
+
+        private void runTest(bool isError, String userName, ITestOutputHelper _output)
+        {
```

Comment:
string parameters and fields should be lowercase string and not capitalized String